connections/models/person.py

```diff
@@ -9,3 +10,15 @@ class Person(Model, CRUDMixin, CreatedUpdatedMixin):
     date_of_birth = db.Column(db.Date, nullable=False)
 
     connections = db.relationship('Connection', foreign_keys='Connection.from_person_id')
+
+    def mutual_friends(self, contact):
+        """Given another person *contact*, return a set of people who are friends with both *self*
+        and *contact*."""
+
+        friends = [conn.to_person_id for conn in self.connections
+                   if conn.connection_type == ConnectionType.friend]
+        contact_fiends = [conn.to_person_id for conn in contact.connections
+                          if conn.connection_type == ConnectionType.friend]
+
+        return [db.session.query(Person).filter(Person.id == friend).first()
+                for friend in set(friends).intersection(set(contact_fiends))]
```

Comment:
The use of Sets here doesn't add anything to the current implementation, but I think it is correct. That said, if Sets were almost never used by the team, I probably wouldn't introduce their use right here. :)